dali/operators/image/remap/rotate_params.h

```diff
@@ -264,23 +264,59 @@ class RotateParamProvider
   }
 
   void InferSize() override {
-    InferSize(std::integral_constant<int, spatial_ndim>());
+    assert(sequence_extents_);
+    const auto &sequence_extents = *sequence_extents_;
+    VALUE_SWITCH(sequence_extents.sample_dim(), ndims_unfolded, (0, 1),
+      (InferSize(sequence_extents.template to_static<ndims_unfolded>());),
+      (DALI_FAIL("Unsupported number of frame extents")));
   }
 
-  void InferSize(std::integral_constant<int, 2>) {
-    assert(static_cast<int>(out_sizes_.size()) == num_samples_);
-    for (int i = 0; i < num_samples_; i++) {
-      auto in_shape = kernels::skip_dim<2>(input_shape_[i]);
-      out_sizes_[i] = RotatedCanvasSize(in_shape, deg2rad(angles_[i]));
+  template <int ndims_unfolded>
+  void InferSize(const TensorListShape<ndims_unfolded> sequence_extents) {
+    assert(sequence_extents.num_elements() == num_samples_);
+    assert(out_sizes_.size() == num_samples_);
+    constexpr auto ndim_tag = std::integral_constant<int, spatial_ndim>();
+    for (int frame_idx = 0, seq_idx = 0; seq_idx < sequence_extents.num_samples(); seq_idx++) {
+      auto num_frames = volume(sequence_extents[seq_idx]);
+      if (num_frames == 0) {
+        continue;
+      }
+      ivec<spatial_ndim> acc_parity;
+      SpatialShape acc_shape;
+      std::tie(acc_shape, acc_parity) = InferSize(ndim_tag, frame_idx);
+      // acc_shape is (extentwise) max of output shapes of all frames in the sample
+      for (int i = 1; i < num_frames; i++) {
+        ivec<spatial_ndim> parity;
+        SpatialShape shape;
+        std::tie(shape, parity) = InferSize(ndim_tag, frame_idx + i);
+        acc_parity += parity;
+        for (int dim_idx = 0; dim_idx < spatial_ndim; dim_idx++) {
+          acc_shape[dim_idx] = std::max(acc_shape[dim_idx], shape[dim_idx]);
+        }
+      }
+      // do the correction of shape extents parity by a majority vote, so that at least half
+      // of the frames in the sequence have the desired parity
+      for (int dim_idx = 0; dim_idx < spatial_ndim; dim_idx++) {
+        bool should_be_odd = 2 * acc_parity[dim_idx] > num_frames;
+        if (acc_shape[dim_idx] % 2 != should_be_odd) {
+          acc_shape[dim_idx]++;
+        }
+      }
```

Comment:
if `acc_shape` was a vec, you could do:
```suggestion
      acc_shape += (acc_shape[dim_idx] % 2) ^ (2 * acc_parity > num_frames);
```

Reply:
done